pkg/controller/job/job_controller.go

```diff
@@ -1041,11 +1042,13 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 			break
 		}
 	}
-	if len(newSucceededIndexes) > 0 {
+
+	if isIndexed {
```

Comment:
Can you please create a separate PR for the changes to job_controller.job and job_controller_test.job so we can patch it back.

Reply:
Done